### ISSUE TITLE
Can't paste a comment (`#`) after a pipe (`|`) and a newline 

### ISSUE THREAD

**Issue by iRon7** [PowerShell/PowerShell#18022]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Paste the following 3 lines into PowerShell (same for Windows PowerShell and VS Code):
```PowerShell
1..3 |
# I do 't expect an error here
ForEach-Object { $_ }
```

### Expected behavior

No errors, the commented line (`# ...`) to be neglected.
Similar to putting the comment directly after the pipe character:

```PowerShell
1..3 | # I don 't expect an error here
ForEach-Object { $_ }
```

### Actual behavior

Error

### Error details

```console
>> # I don 't expect an error here
ParserError:
Line |
   1 |  1..3 |
     |        ~
     | An empty pipe element is not allowed.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.6
PSEdition                      Core
GitCommitId                    7.2.6
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

<img width="252" alt="2022-09-03_19h55_01" src="https://user-images.githubusercontent.com/8080834/188282675-bb297e19-5785-4e58-bfdf-ab512a097dd6.png">

**Comment by dkaszews** [comment]
I think it's broken line continuation after a comment. I can reproduce this simpler case for any trailing binary operator:

```pwsh
> 1 +
>> 2
3
> 1 +
>> # comment
ParserError:
Line |
   1 |  1 +
     |     ~
     | You must provide a value expression following the '+' operator.
```

In all cases I am doing shift+enter to continue the line, but after the comment it treats it like I just typed enter.

Interestingly, it works without problem in a script:
```pwsh
> @"
>> 1 +
>> 2
>> "@ | Invoke-Expression
3
> @"
>> 1 +
>> # comment
>> 2
>> "@ | Invoke-Expression
3
```

So my guess is that one should look like at code handling shift+enter or pasted code specifically, not the interpreter itself.

**Comment by 237dmitry** [comment]
It seems to me that a similar issue has already been. Discussed workaround like:
```powershell
PS > & {
1..3 |
# I do 't expect an error here
ForEach-Object { $_ }
}
```

**Comment by dkaszews** [comment]
@237dmitry Doesn't that create a separate scope? So that `$x = 3; & { $x = 5 }; $x | Should -Be 5` will fail?

**Comment by 237dmitry** [comment]
> Doesn't that create a separate scope?

Yes, this is separate scope. IMHO, the most comfortable construction to script in the console in order to not killing variables.

**Comment by kilasuit** [comment]
This is dependant on how you paste to the console & where you have it running

In Windows Terminal
Neither `CTRL + V`, `CTRL + SHIFT + V`, `SHIFT + INSERT`,`` or right click paste worked for me

Not in Windows Terminal, i.e. in a pwsh spawned console
right click paste seems to fail whilst for me where as CTRL + V worked fine

In VSCode
right click paste seems to fail whilst for me where as CTRL + V worked fine

**Comment by kilasuit** [comment]
I think this is how this should work as PSReadline gave us right click paste as well as the existing CTRL + V paste method, and they have always worked differently 

From an WG Interactive Console & PSReadline perspective I think this is as it should be & don't think this needs review by the Engine WG but will let them review

**Comment by 237dmitry** [comment]
Couldn't reproducible this issue in Linux (tilix terminal).

**Comment by dkaszews** [comment]
Can reproduce this on Linux (Chromebook terminal ssh'ed into Ubuntu)

**Comment by jhoneill** [comment]
> > Doesn't that create a separate scope?
> 
> Yes, this is separate scope. IMHO, the most comfortable construction to script in the console in order to not killing variables.

Or use `.` instead of `&`

**Comment by mklement0** [comment]
As an aside: It is a separate issue, but it is curious that parsing stops _after the comment line_ and considers the statement complete, given that it would make more sense to keep reading, based on the pending `|`.

A more example would be to try to past the following (PS Core now allows placing the `|` on the _next_ line):

```powershell
# Breaks when pasted with right-clicking on Windows and in Unix terminals
1..3
  | Write-Output
```

**Comment by iRon7** [comment]
The issue still exists in `7.4.2` terminal with a right click:

![image](https://github.com/PowerShell/PowerShell/assets/8080834/00b101c0-d0bd-4b76-95ff-f7d1c1582ebd)

but I am happy if this issue is closed with <kbd>won't fix</kbd>.

**Comment by iRon7** [comment]
The issue still exists in:

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.4.6
PSEdition                      Core
GitCommitId                    7.4.6
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iRon7** [comment]
The issue still exists in `7.5.1`

**Comment by xtqqczze** [comment]
@kilasuit Please add label `KeepOpen`.

**Comment by iRon7** [comment]
The issue still exists in `7.5.4`

**Comment by kilasuit** [comment]
I looked at this again quickly and whilst I agree that this is a bug with how pasting works when PSReadline is loaded as [@mklement0 mentioned above](https://github.com/PowerShell/PowerShell/issues/18022#issuecomment-1239474131) this I think would need fixed in the PSReadLine Repo even though this then bubbles up as a parser error.

This is currently assigned to @daxian-dbw but apparently I can also assign to Copilot so lets give it a try.

**Comment by kilasuit** [comment]
I tried to but got an error in doing so likely permissions related so maybe @daxian-dbw / @theJasonHelmick  can assign this to Copilot to investigate?
